### PR TITLE
Fix/repair location services

### PR DESCRIPTION
**Pull request by colinsmith**
This PR fixes a bug with how custom location services are passed in via options.

Currently, if you provide a custom `serviceDefinitions: {}` function, it throws an error due to only having `defaultOptions` accessible in `Locations.js`.

Additionally, when `isScript` is set to `true`, there is a typo in the object the async helper is trying to access. This is resolved in this PR as well.

With this change a custom `serviceDefinition` and `service` no longer throws an error, for example:

```
const cc = new CC({
    ...options
    location: {
        serviceDefinitions: {
            myService: function(options) {
                return {
                    url: 'example/myScript.js',
                    isScript: true,
                    callback: function () {
                        return true;
                    }
                }
            },
        },
        services: ['myService'],
    }
});
```

### CODE REVIEW

File: src/models/Location.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { isPlainObject, getScript, makeAsyncRequest } from "../utils"
 export default class Location extends Base {
   constructor( options ) {
     super( defaultOptions, options )
+    this.options = { ...defaultOptions, ...options }; // combine default options and user options
```
Comment (przemyslawzalewski) [inline src/models/Location.js]:
```suggestion
    this.options = { ...defaultOptions, ...(options || {}) }; // combine default options and user options
```

I'm good with that as the [`super`s merging behavior ](https://github.com/sandstreamdev/cookieconsent/blob/dev/src/models/Base.js#L10) is inappropriate for this case. I just wonder why this [particular test case passes](https://github.com/sandstreamdev/cookieconsent/blob/dev/src/models/Location.spec.js#L33) when it should fail when options are missing (object spread of `undefined`)? 🤔

**Review by przemyslawzalewski — APPROVED** [review]
Thank you for your deep investigation with the detailed description and for submitting the fixes! Kudos for that 👍

File: src/models/Location.js (hunk 1 above)
Comment (colinsmith) [reply src/models/Location.js]: Thanks for this!

I've gone and updated my PR with your suggestion, as well as added another test case to cover passing custom `serviceDefinitions` options. This should help catch this sort of error in the future 😄 

From what I can tell, the existing test case was only validating the structure of the object (`isLocation`) and the merged `options` object keys (`hasCorrectOptions`).

With the way `Locations.js` is written, the `getServiceByIdx()` function needs to be called to actually cause an error. The error is caused when it goes to access `this.options.serviceDefinitions[...]`, and the custom object key does not exist.

**Comment by przemyslawzalewski** [comment]
Thank you for once again! I will create a patch release for this. Stay tuned!

**Comment by przemyslawzalewski** [comment]
@colinsmith You change has landed on [v1.0.1](https://www.npmjs.com/package/@sandstreamdev/cookieconsent/v/1.0.1) with the following changes https://github.com/sandstreamdev/cookieconsent/compare/sandstreamdev:1e39fa63a9b7d3af46659722285e8b61e04060d9...d7e7a76d2154bc05fc1df3cc175f945dded9d1d1 🚀

**Pull request merged**